Atuadores/Potenciometro_Motorizado-PWM-PID/README.md

```diff
@@ -0,0 +1,368 @@
+}
+```
+
+Por fim, o último callback foi utilizado para ler o valor atual do potenciômetro, e enviá-los para a fila xQueueAFEC.
+
+```c
+QueueHandle_t xQueueAFEC;
+
+#define AFEC_POT AFEC0
+#define AFEC_POT_ID ID_AFEC0
+#define AFEC_POT_CHANNEL 0 // Canal do pino PD30
+
+static void afec_callback(void) {
+	uint value;
+	value = afec_channel_get_value(AFEC_POT, AFEC_POT_CHANNEL);
+	BaseType_t xHigherPriorityTaskWoken = pdTRUE;
+	xQueueSendFromISR(xQueueAFEC, &value, &xHigherPriorityTaskWoken);
+}
+```
+
+# Definindo as tasks
+
+Foram utilizadas duas tasks nesse módulo. A task_afec mexe com os valores de afec e com o OLED, e a task_motor usa os valores recebidos da task_afec pela fila xQueueMOTOR para alterar o valor do potenciômetro para o valor desejado usando PWM controlado por PID. 
+
+Na task_afec, primeiro se inicializa o rtt e a tela daplaca OLED, já escrevendo o valor 4095, que é o valor determinado como default do valor desejado para o potenciômetro. Em seguida, foram determinados os valores controlados pelos botões da placa OLED que serão os valores alvos para os quais o potenciômetro terá que se ajustar. Por fim, foi feito um loop infinito, onde o valor lido pelo afec é armazenado em uma variável de tipo struct potenciometro (em valor_atual), o modo determinado pelo botão pressionado é usado para determinar o valor armazenado na mesma variável de tipo struct potenciometro (em valor_desejado), e por fim a variável de tipo struct é enviada pela fila xQueueMOTOR para a task_motor.
+
+```c
+QueueHandle_t xQueueMOTOR;
+
+typedef struct {
+	uint valor_atual;
+	uint valor_desejado;
+} potenciometro;
+
+static void task_afec(void *pvParameters) {
+	RTT_init(1000, 100, RTT_MR_ALMIEN);
+	config_AFEC_pot(AFEC_POT, AFEC_POT_ID, AFEC_POT_CHANNEL, afec_callback);
+	
+	gfx_mono_ssd1306_init();
+	
+	gfx_mono_draw_filled_rect(0, 0, 150, 20, GFX_PIXEL_CLR);
+	gfx_mono_draw_string("4095", 0,0, &sysfont);
+
+	char nome1[]="10";
+	int val1 = 10;
+	
+	char nome2[]="2000";
+	int val2 = 2000;
+	
+	char nome3[]="4095";
+	int val3 = 4095;
+	
+	char *valores_potenciometro[]={&nome1, &nome2, &nome3};
+
+	uint valor_afec;
+	uint valor_oled;
+	
+	potenciometro pot1;
+	pot1.valor_desejado = 4095;
+
+	
+	for (;;) {
+		if (xQueueReceiveFromISR(xQueueAFEC, &valor_afec, 10)){
+			
+			pot1.valor_atual = valor_afec;
+			
+			
+		} 
+		
+		if (xQueueReceiveFromISR(xQueueModo, &valor_oled, 10)){
+			gfx_mono_draw_filled_rect(0, 0, 150, 20, GFX_PIXEL_CLR);
+			gfx_mono_draw_string("entrou", 0,0, &sysfont);
+			char *valor_potenciometro = valores_potenciometro[valor_oled-1];
+			
+			gfx_mono_draw_filled_rect(0, 0, 150, 20, GFX_PIXEL_CLR);
+			gfx_mono_draw_string(valor_potenciometro, 0,0, &sysfont);
+			
+			if (valor_oled == 1){
+				pot1.valor_desejado = val1;
+			}
+			if (valor_oled == 2){
+				pot1.valor_desejado = val2;
+			}
+			if (valor_oled == 3){
+				pot1.valor_desejado = val3;
+			}
+		}
+		
+		xQueueSend(xQueueMOTOR, (void *)&pot1, 10);
+
+	}
+}
+```
```

Comment:
tirar.. não tem sentido colocar muito código assim, só se for explicar e colocar trechos 
```suggestion

```